server/src/main/java/org/elasticsearch/ingest/PipelineConfiguration.java

```diff
@@ -121,8 +125,42 @@ public String toString() {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(id);
-        out.writeBytesReference(config);
-        XContentHelper.writeTo(out, xContentType);
+        final BytesReference bytesReferenceToWrite;
+        final XContentType xContentTypeToWrite;
+        if (out.getVersion().before(Version.V_8_0_0)) {
+            /*
+             * While serialization works fine regardless of which fields are in the PipelineConfiguration, the Pipeline class throws an
+             * exception if there are fields in the PipelineConfiguration that it does not expect. So when sending the config to older
+             * versions we have to strip out newer fields.
+             */
```

Comment:
Can you explain why we need this check if we're disallowing in `PutPipelineTransportAction` the creation of an ingest pipeline with a `_meta` field unless all nodes in the cluster support it?

Reply:
Hmm, I had added this before I added that check in PutPipelineTransportAction, and then hadn't thought about it again. I'm actually not sure when this serialization would be used across nodes since each node pulls the config from the cluster state. If an old node were to come online after being down during the PutPipelineTransportAction call, it would pull from the cluster state (and blow up, as we want) rather than going through this serialization, right?

Reply:
We had a couple of conversations in slack. The only time that this code would be relevant is if an older node were not in the cluster when the pipeline was created, but then came back into the cluster as master. Fortunately @DaveCTurner tells us that this situation can no longer happen as of https://github.com/elastic/elasticsearch/pull/73358 . So I'll remove this check altogether.